src/http-adapter.js

```diff
@@ -82,10 +85,34 @@ class HttpAdapter {
       'Content-Type': 'application/vnd.api+json'
     }
 
+    // Stringify data before any async stuff, just in case it's accidentally a mutable object (e.g.
+    // some instrumented Vue data)
```

Comment:
I think the data must be stringified here because `application/vnd.api+json` does not trigger jQuery's automatic type detection and do this for us.